ministryofjustice/laa-assess-crime-forms
# updated azure registration details and refer to new secret keys in yaml

##

 [CRM457-1230](https://dsdmoj.atlassian.net/browse/CRM457-1230)

Previously the caseworker application was using credentials and secrets from the app store azure registration to verify it's identity with the app store.

It _should_ have been using it's own client registration. As can be seen in the [example documentation](https://intility.github.io/fastapi-azure-auth/usage-and-faq/calling_your_apis_from_python#single--and-multi-tenant), the scope uses a different client_id. 

Doing it this way will also allow us to use different roles for provider and caseworker once they have been implemented by Azure.

https://portal.azure.com/#view/Microsoft_AAD_RegisteredApps/ApplicationsListBlade
 
Two new azure registrations has been created for the provider application. 
`laa-crime-forms-caseworker-datastore-access [pre-production]` and `laa-crime-forms-caseworker-datastore-access [production]`

Client secrets are created in pre-production azure registration for the dev and uat environments and in the production azure environment for prod.


## Notes for reviewer

For this code to work - the `app-store-auth` secret needs to be updated in kubernetes to include the new keys.
That has been done - the old keys currently exist and should be deleted when this has been merged.

## Screenshots of changes (if applicable)

### Before changes:

### After changes:

## How to manually test the feature
You can check this works by adding in the correct environment variables to your env.local files and submitting an application. You should create your own secret in the azure registration.


[CRM457-1287]: https://dsdmoj.atlassian.net/browse/CRM457-1287?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

[CRM457-1230]: https://dsdmoj.atlassian.net/browse/CRM457-1230?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
